java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -893,21 +893,55 @@ private static JavaType getPrimitive(JavaType primitiveOrWrapper) {
   }
 
   public List<JavaType> findSamMethodArgs(Type type) {
-    for (Symbol member : type.symbol().memberSymbols()) {
-      if(member.isMethodSymbol() && member.isAbstract()) {
-        JavaSymbol.MethodJavaSymbol target = (JavaSymbol.MethodJavaSymbol) member;
-        List<JavaType> argTypes = ((MethodJavaType) target.type).argTypes;
-        argTypes = typeSubstitutionSolver.applySiteSubstitutionToFormalParameters(argTypes, (JavaType) type);
-        return argTypes.stream().map(argType -> {
-          if (argType.isTagged(JavaType.WILDCARD)) {
-            // JLS8 9.9 Function types : this is approximated for ? extends X types (cf JLS)
-            return ((WildCardType) argType).bound;
-          }
-          return argType;
-        }).collect(Collectors.toList());
+    return findSamMethodArgsRecursively(type).orElse(new ArrayList<>());
+  }
+  
+  private Optional<List<JavaType>> findSamMethodArgsRecursively(@Nullable Type type) {
+    if(type == null) {
+      return Optional.empty();
+    }
+    Optional<List<JavaType>> result = type.symbol().memberSymbols().stream()
+      .filter(Resolve::isAbstractMethod).findFirst()
+      .map(s -> ((MethodJavaType) ((JavaSymbol.MethodJavaSymbol) s).type).argTypes);
+
+    if(!result.isPresent()) {
+      result = findSamMethodArgsRecursively(type.symbol().superClass());
+      if(!result.isPresent()) {
+        result = type.symbol().interfaces().stream().map(this::findSamMethodArgsRecursively).filter(Optional::isPresent).map(Optional::get).findFirst();
```

Comment:
@tvsonar this warning comes from Dory and is raised when we fail to resolve a method (revealing a flaw in semantic)